components/pendingRewards.ts

```diff
@@ -0,0 +1,96 @@
+import { PrismaClient } from '@prisma/client';
+import { Assets, Blockfrost, C, Lucid, Network, Tx, TxComplete, Unit, UTxO, utxoToCore } from 'lucid-cardano';
+import dotenv from 'dotenv';
+import sample from 'lodash.sample'
+import stakingpools from '../resources/stakingPools.js'
+import { addAssets, subAssetsFromUtxos, divideAssetsBy2 } from '../util/sc.js'
+import { dbFromPool } from '../util/snapshot.js'
+dotenv.config();
+
+const BLOCKFROST_URL = process.env.BLOCKFROST_URL
+const BLOCKFROST_KEY = process.env.BLOCKFROST_API_KEY
+const NETWORK = process.env.BLOCKFROST_NETWORK
+
+if (!BLOCKFROST_URL || !BLOCKFROST_KEY || !NETWORK) {
+  throw 'Environment variables not set'
+}
+
+type RewardReqBody = {
+  poolIndex: number,
+  address: string
+}
+
+const pendingRewardsCtrl = async (prisma: PrismaClient, body: string) => {
+    const parsedBody: RewardReqBody = JSON.parse(body)
+    let poolI = parsedBody.poolIndex
+    let address = parsedBody.address
+    const poolargs = stakingpools[poolI]
+
+    let pool = await prisma.pool.findFirst({
+        where: {
+            poolIndex: poolI
+        }
+    })
+    if (!pool) {
+        pool = await prisma.pool.create({
+            data: {
+                poolIndex: poolI
```

Comment:
Did you change any code in the pendingRewards or submit controllers, or did you just relocate the controllers into different files?

Reply:
Ah I see, you created them new as v2. Are there any differences between the old controllers? If yes very hard to recognize any changes compared to the old controllers (that was the reason for my question)